### PR TITLE
feat: task cis-to-catalog

### PR DESCRIPTION
**Pull request by degenaro**
## Description

Add new trestle task cis-to-catalog that takes input file(s) cis.profile, cis-node.profile (from ComplianceAsCode) and transforms into OSCAL output file catalog.json.

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar

### CODE REVIEW

File: trestle/tasks/cis_to_catalog.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        root = Group(title='root', groups=[])
+        for node in root_nodes:
+            group = Group(title=f'{node.name} {node.description}')
+            root.groups.append(group)
+            depth = self._depth(node.name)
+            if depth == 3:
+                self._add_groups(group, node.name, depth)
+            if depth == 2:
+                self._add_controls(group, node.name, depth)
+        # metadata
```
Comment (fsuits) [inline trestle/tasks/cis_to_catalog.py:158]: Is it possible controls might not be in a group and instead should be in a controls attribute of the catalog itself?

**Review by fsuits — APPROVED** [review]
LGTM - just had one small question

File: trestle/tasks/cis_to_catalog.py (hunk 1 above)
Comment (degenaro) [reply trestle/tasks/cis_to_catalog.py:158]: So far, this has not been the case. If that changes we can adjust the code accordingly.

File: trestle/tasks/cis_to_catalog.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+    @property
+    def name(self) -> str:
+        """Node name."""
+        return self._name
+
+    @property
+    def description(self) -> str:
+        """Node description."""
+        return self._description
```
Comment (butler54) [inline trestle/tasks/cis_to_catalog.py:55]: For something like this why don't we just use a pydantic class (off of BaseModel not OscalBaseModel)?

**Review by butler54 — APPROVED** [review]
LGTM - see the comment for your optional revision.

File: trestle/tasks/cis_to_catalog.py (hunk 2 above)
Comment (degenaro) [reply trestle/tasks/cis_to_catalog.py]: Fixed.

**Pull request merged**